includes/kco-functions.php

```diff
@@ -680,6 +680,81 @@ function kco_validate_order_total( $klarna_order, $order ) {
 	return true;
 }
 
+/**
+ * Validate that the Woo order matches the corresponding Klarna order.
+ *
+ * @param array    $klarna_order The Klarna order.
+ * @param WC_Order $order The Woo order.
+ *
+ * @return bool
+ */
+function kco_validate_order_content( $klarna_order, $order ) {
+	$order_data = new KCO_Request_Order();
+
+	$mismatch = false;
+	foreach ( $order->get_items() as $order_item ) {
+		$order_line = $order_data->get_order_line_items( $order_item );
+		if ( $mismatch ) {
+			break;
+		}
+
+		$match     = false;
+		$reference = $order_line['reference'];
+		foreach ( $klarna_order['order_lines'] as $klarna_order_item ) {
+			$match = false;
+			if ( $reference === $klarna_order_item['reference'] ) {
+				$match = true;
+				if ( $klarna_order_item['quantity'] !== $order_line['quantity'] ) {
+					$mismatch = true;
+				}
+
+				break;
+			}
+		}
+
+		// Check if the Woo item was not found in the Klarna order.
+		if ( ! $match ) {
+			$mismatch = true;
+		}
+	}
+
+	$shipping        = ! empty( $order->get_shipping_method() ) ? $order_data->get_order_line_shipping( $order ) : false;
+	$klarna_shipping = array_filter(
+		$klarna_order['order_lines'],
+		function ( $order_line ) {
+			return 'shipping_fee' === $order_line['type'];
+		}
+	);
+	$klarna_shipping = reset( $klarna_shipping );
+
+	if ( ! empty( $klarna_shipping ) && empty( $shipping ) ) {
+		$mismatch = true;
+	} elseif ( empty( $klarna_shipping ) && ! empty( $shipping ) ) {
+		$mismatch = true;
+	} else {
+		// If KSA is enabled, we'll skip the control.
+		$is_ksa = strpos( $shipping['reference'], 'klarna_kss' ) !== false;
+		if ( ! $is_ksa && $shipping['reference'] !== $klarna_shipping['reference'] ) {
+			$mismatch = true;
+		}
```

Comment:
If both `$klarna_shipping` and `$shipping` is empty here, cant we get to the case where we try to get the reference from both of the items that might not have this key causing a error?